executor/chroot/chroot_executor.go

```diff
@@ -91,15 +89,29 @@ func (e *Executor) Execute(f def.Formula, j def.Job, d string) def.JobResult {
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
 
-	if err := cmd.Start(); err != nil {
-		if err2, ok := err.(*exec.Error); ok && err2.Err == exec.ErrNotFound {
+	// launch execution.
+	// transform gosh's typed errors to repeatr's hierarchical errors.
+	var proc gosh.Proc
+	try.Do(func() {
+		proc = gosh.ExecProcCmd(cmd)
+	}).CatchAll(func(err error) {
+		switch err.(type) {
+		case gosh.NoSuchCommandError:
 			panic(executor.NoSuchCommandError.Wrap(err))
+		case gosh.NoArgumentsErr:
+			panic(executor.NoSuchCommandError.Wrap(err))
+		case gosh.ProcMonitorError:
+			panic(executor.TaskExecError.Wrap(err))
+		default:
+			panic(executor.UnknownError.Wrap(err))
 		}
-		panic(executor.TaskExecError.Wrap(err))
-	}
+		// FIXME all *this* also should actually be done in the barrier before returning a job, methinks?
+		// actually, not clear.  we haven't really spec'd that.  REVIEW.
+	}).Done()
 
 	// Wait for the job to complete
-	result.ExitCode = flak.WaitAndHandleExit(cmd)
+	// REVIEW: consider exposing `gosh.Proc`'s interface as part of repeatr's job tracking api?
+	result.ExitCode = proc.GetExitCode()
```

Comment:
I think we might end up more in the business of implementing something that looks very like the [Proc interface](https://godoc.org/github.com/polydawn/gosh#Proc), honestly. I want a progress-report concept of state that has more steps than just [these states](https://godoc.org/github.com/polydawn/gosh#State) for example.

Oh and by the way we have one struct to report on this but really the Executor ought to stay in control of that. More fuel for my thought of the Executor gaining a method to build itself a Job way before the Scheduler wants to run a Formula, and the Job gets better at saying what state it's in.

I propose that, ignoring gosh, we figure out how a running Job should really look, and then decide what parts (if any) of Gosh bleed through. I might have a dedicated branch to poke at that after we "clear the board" a bit.